### PR TITLE
Implement some todos.

### PR DESCRIPTION
**Pull request by ptersilie**

### CODE REVIEW

File: ykcompile/src/lib.rs [hunk 1]
```diff
@@ -975,7 +975,8 @@ impl<TT> TraceCompiler<TT> {
             Location::NotLive => unreachable!(),
         }
         let dest_loc = self.iplace_to_location(dest);
-        self.store_raw(&dest_loc, &*TEMP_LOC, SIR.ty(&src.ty()).size());
+        debug_assert_eq!(SIR.ty(&dest.ty()).size(), *PTR_SIZE);
+        self.store_raw(&dest_loc, &*TEMP_LOC, *PTR_SIZE);
```
Comment (vext01) [inline ykcompile/src/lib.rs:979]: This was actually a bug. Might it be worth putting it in a separate commit with an explanation?

File: ykcompile/src/lib.rs [hunk 2]
```diff
@@ -1429,7 +1431,37 @@ impl<TT> TraceCompiler<TT> {
...
+                        4 => dynasm!(self.asm
+                            ; mov Rq(*TEMP_REG), QWORD [Rq(src_ro.reg) + src_ro.off]
+                            ; push Rq(src_ro.reg)
+                            ; mov Rq(src_ro.reg), QWORD [Rq(dest_ro.reg) + dest_ro.off]
+                            ; mov DWORD [Rq(src_ro.reg) + *dest_off], Rd(*TEMP_REG)
+                            ; pop Rq(src_ro.reg)
+                        ),
+                        8 => dynasm!(self.asm
+                            ; mov Rq(*TEMP_REG), QWORD [Rq(src_ro.reg) + src_ro.off]
+                            ; push Rq(src_ro.reg)
```
Comment (vext01) [inline ykcompile/src/lib.rs:1458]: Genius! I didn't know what you meant when you explained in MM, but now I do!

File: ykcompile/src/lib.rs [hunk 3]
```diff
@@ -1466,7 +1498,16 @@ impl<TT> TraceCompiler<TT> {
                         ; jne ->guardfail
                     );
                 }
-                Location::Mem(..) => todo!(),
+                Location::Mem(ro) => {
+                    debug_assert!(SIR.ty(&val.ty()).size() <= 8);
```
Comment (vext01) [inline ykcompile/src/lib.rs:1502]: That `8` is a bit of a magic number. Shall we make a const for it?

Comment (ptersilie) [reply ykcompile/src/lib.rs:1502]: Oh, that's left over from our debugging sessions. I don't think we need this here.

File: ykcompile/src/lib.rs (hunk 1 above)
Comment (ptersilie) [reply ykcompile/src/lib.rs:979]: Can do, but will have to force push.

**Comment by ptersilie** [comment]
Let me know if okay to squash and then I'll split this up into two commits as discussed above.

**Comment by vext01** [comment]
Go ahead

**Comment by ptersilie** [comment]
Squashed and split up the commits.

**Comment by ptersilie** [comment]
Try again.
